### PR TITLE
Script to add release notes to docs-website

### LINKED ISSUE #590: Automate release notes for docs site
Unsure if possible but worth researching and pushing to see if this can become possible.

Another option is having these docs link back to GitHub release notes. Explore various options toward reducing this toil.

Prob should chat with the docs team as I believe they have some work in progress.

### PR DESCRIPTION
**Pull request by bizob2828**
## Proposed Release Notes
 * Added a script to be used by agent developers to add a PR to `docs-website` after the release of agent.

## Links
Closes #590

## Details
Currently this relies on the fact you have a fork of docs-website on your local machine.  Once we get a machine user we can remove some code around specifying the username as we will be cloning the repo in CI and adding PR from the repo itself.

To run script:

```sh
GITHUB_TOKEN=<redacted> node bin/create-docs-pr.js --username bizob2828 --tag v8.7.0 --repo-path /path/to/docs-website
```

You can also use the `--dry-run` flag to run through script but it will not create the branch, push to origin and create PR.

### CODE REVIEW

File: bin/create-docs-pr.js [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+program.option('--dry-run', 'executes script but does not commit nor create PR')
+program.option(
+  '--repo-path <path',
+  'Path to the docs-website fork on local machine',
+  'docs-website'
+)
+const RELEASE_NOTES_PATH =
+  './src/content/docs/release-notes/agent-release-notes/nodejs-release-notes'
+
+async function createRelease() {
```
Comment (michaelgoin) [inline bin/create-docs-pr.js:37]: Would prob rename this function to be something like `createReleaseNotePr` or something more specific to the task.

File: bin/create-docs-pr.js [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+'use strict'
+
+const fs = require('fs')
+const path = require('path')
+const { program } = require('commander')
+
+const Github = require('./github')
+const git = require('./git-commands')
+
+const DEFAULT_FILE_NAME = 'NEWS.md'
```
Comment (michaelgoin) [inline bin/create-docs-pr.js:15]: A thought that if we eventually trigger off of say a successful release publish, we could also just grab from that release in GitHub. Then wouldn't have to parse down the text / duplicate the work that we do to create the GitHub release in the first place. May or may not be the way to go but food for thought.

File: bin/create-docs-pr.js [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+    logStep('Validation')
+    validateTag(version, options.force)
+    logStep('Get Release Notes from File')
+    const { body, releaseDate } = await getReleaseNotes(version, options.changelog)
+    logStep('Branch Creation')
+    const branchName = await createBranch(options.repoPath, version, options.dryRun)
+    logStep('Format release notes file')
+    const releaseNotesBody = formatReleaseNotes(releaseDate, version, body)
+    logStep('Create Release Notes')
+    await addReleaseNotesFile(releaseNotesBody, version)
```
Comment (michaelgoin) [inline bin/create-docs-pr.js:58]: Hard to tell just reading the code but seems like in this case the version would be `Major.minor.patch`. For the file naming convention, we want `Major-minor-patch` which doesn't seem like that conversion happens anywhere. But I may just not be interpreting things right

File: bin/create-docs-pr.js [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+async function commitRelaseNotes(version, remote, branch, dryRun) {
+  if (dryRun) {
+    console.log('Dry run indicated (--dry-run), skipping committing release notes.')
+    return
+  }
+
+  console.log(`Adding release notes for ${version}`)
+  const files = [getFileName(version)]
+  await git.addFiles(files)
+  await git.commit(`chore: Node.js Agent ${version} Release Notes.`)
```
Comment (michaelgoin) [inline bin/create-docs-pr.js:234]: 👍 conventional commit covered.

Comment (michaelgoin) [reply bin/create-docs-pr.js:234]: Minor: Might insert an `"Adds"` before "Node.js Agent".

File: bin/create-docs-pr.js [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+ */
+async function createPR(username, version, branch, dryRun) {
+  if (!process.env.GITHUB_TOKEN) {
+    console.log('GITHUB_TOKEN required to create a pull request')
+    stopOnError()
+  }
+
+  const github = new Github('newrelic', 'docs-website')
+  const title = `Node.js Agent ${version} Release Notes`
+  const body = [
```
Comment (michaelgoin) [inline bin/create-docs-pr.js:255]: You could prob get rid of most of the formatting here if really wanted. So far, I've typically put minimal messaging (sometimes nothing). https://github.com/newrelic/docs-website/pull/4287

File: bin/create-docs-pr.js (hunk 2 above)
Comment (bizob2828) [reply bin/create-docs-pr.js:15]: true. i just did this because we already had helpers to parse the file

File: bin/create-docs-pr.js (hunk 3 above)
Comment (bizob2828) [reply bin/create-docs-pr.js:58]: oh, good catch. yes currently it is `major.minor.patch` ill change to be `major-minor-patch`

**Review by michaelgoin — APPROVED** [review]
👍

**Pull request merged**